layers/meta-balena-raspberrypi/recipes-core/images/balena-image.bbappend

```diff
@@ -28,7 +28,12 @@ python overlay_dtbs_handler () {
     for soc_fam in d.getVar('SOC_FAMILY', True).split(':'):
         if soc_fam == 'rpi':
             resin_boot_partition_files = d.getVar('BALENA_BOOT_PARTITION_FILES', True)
-
+            # All changes to KERNEL_DEVICETREE are local and cannot be modified
+            # globally, thus we save the list of existing overlays when they are built
+            f = open(d.getVar('DEPLOY_DIR_IMAGE') + '/overlays.txt', "r")
```

Comment:
Currently the PR fails with:

```
Parsing recipes...ERROR: /work/build/../layers/meta-balena/meta-balena-common/recipes-core/images/balena-image.bb: Execution of event handler 'defaultoverlay_dtbs_handler' failed
Traceback (most recent call last):
  File "/work/build/../layers/meta-balena-raspberrypi/recipes-core/images/balena-image.bbappend", line 33, in defaultoverlay_dtbs_handler(e=<bb.event.RecipePreFinalise object at 0x7fab3921df98>):
                 # globally, thus we save the list of existing overlays when they are built
    >            f = open(d.getVar('DEPLOY_DIR_IMAGE') + '/overlays.txt', "r")
                 kernel_devicetree = f.read()
FileNotFoundError: [Errno 2] No such file or directory: '/work/build/tmp/deploy/images/npe-x500-m3/overlays.txt'
```

Maybe you should use the `a+` mode for `open` instead of `r`?

Reply:
Unfortunately it seems that the file really doesn't exist at that point when this error occurs, right at the start of the yocto build when all recipes are parsed. The file is expected to get populated later, after the kernel sources are downloaded. I need to look further into this.